src/pybind/mgr/dashboard/frontend/src/app/core/mgr-modules/mgr-module-details/mgr-module-details.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, Input, OnChanges } from '@angular/core';
+
+import { MgrModuleService } from '../../../shared/api/mgr-module.service';
+import { CdTableSelection } from '../../../shared/models/cd-table-selection';
+
+@Component({
+  selector: 'cd-mgr-module-details',
+  templateUrl: './mgr-module-details.component.html',
+  styleUrls: ['./mgr-module-details.component.scss']
+})
+export class MgrModuleDetailsComponent implements OnChanges {
+  module_config: any;
+
+  @Input()
+  selection: CdTableSelection;
+
+  constructor(private mgrModuleService: MgrModuleService) {}
+
+  ngOnChanges() {
+    if (this.selection.hasSelection) {
+      const selectedItem = this.selection.first();
```

Comment:
Why not saving the observable (`this.module_config$`) and leaving the template deal with it with aynscPipe?

Reply:
Yes, this would work, but to keep consistency to the rest of the UI i suggest to keep the current code. Your suggestion looks smarter, but it makes it impossible to debug.

Reply:
Why is that? Sorry, I'm still a newcomer to UI stuff :sweat_smile:

Reply:
If i'm wrong, please correct me (maybe Firefox and Chrome supports the debugging of an observable). In the below code you can easily set a breakpoint in the subscribe function.

```
if (this.selection.hasSelection) {
      const selectedItem = this.selection.first();
      this.mgrModuleService.getConfig(selectedItem.name).subscribe((resp: any) => {
        this.module_config = resp;
      });
    }
```

If you use 
```
this.module_config$ = this.mgrModuleService.getConfig(selectedItem.name);
```
there is no way to debug the result of the ``getConfig()`` method because where should you set a breakpoint when the observable succeeds.

Reply:
Why not `tap(console.debug(...))`?

Reply:
I'm in favor of setting debug break points and inspecting the result in the debug tools and not only in the console.